### PR TITLE
Bump archive version

### PR DESCRIPTION
**Pull request by kobybr**

### CODE REVIEW

File: .sync.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+
+mock_with: ':rspec'
+
```
Comment (edestecd) [inline .sync.yml]: can you add the other items like here:
https://github.com/MiamiOH/puppet-duo_authproxy/blob/master/.sync.yml

Comment (edestecd) [reply .sync.yml]: specifically the gitlab-ci and travis stuff

File: manifests/install.pp [hunk 2]
```diff
@@ -143,6 +143,9 @@
 
   # Fetch and install the ShibCAS component.
   if $include_cas {
+    if $casclient_source == undef {
```
Comment (edestecd) [inline manifests/install.pp]: Why not use validation to check this?

File: manifests/init.pp [hunk 3]
```diff
@@ -84,6 +80,8 @@
   $service_ensure          = $shibboleth_idp::params::service_ensure,
   $service_restart         = $shibboleth_idp::params::service_restart,
 
+  $casclient_source        = undef,
```
Comment (edestecd) [inline manifests/init.pp]:
```suggestion
  String $casclient_source,
```

Comment (edestecd) [reply manifests/init.pp]: Not setting a default enforces that they must set this.

File: manifests/install.pp (hunk 2 above)
Comment (edestecd) [reply manifests/install.pp]: I'd suggest the validation above and remove this if

Comment (kobybr) [reply manifests/install.pp]: Isn't validate deprecated?

File: manifests/install.pp [hunk 4]
```diff
@@ -210,7 +212,7 @@
   # The install should run only once for a given version. This creates
   # the Shibboleth install base (by default /opt/shibboleth-dip).
   exec { 'shibboleth idp install':
-    command     => "${shibboleth_idp::shib_src_dir}/shibboleth-identity-provider-${shibboleth_idp::shib_idp_version}/bin/install.sh  -Didp.property.file=${shibboleth_idp::shib_src_dir}/idp.install.properties",
+    command     => "${shibboleth_idp::shib_src_dir}/shibboleth-identity-provider-${shibboleth_idp::shib_idp_version}/bin/install.sh  -Didp.property.file=${shibboleth_idp::shib_src_dir}/idp.install.properties", # lint:ignore:140chars
```
Comment (edestecd) [inline manifests/install.pp]: many ignore this globally in Rakefile

Comment (edestecd) [reply manifests/install.pp]: https://github.com/puppetlabs/pdk-templates#rakefile
`default_disabled_lint_checks`

File: manifests/params.pp [hunk 5]
```diff
@@ -32,11 +32,7 @@
   $ldap_return_attributes  = []
 
   $slf4j_version           = '1.7.22'
-  $slf4j_checksum_type     = 'md5'
```
Comment (edestecd) [inline manifests/params.pp]: why remove these as parameters?

If you do you may want to list this as a breaking change in the changelog

File: metadata.json [hunk 6]
```diff
@@ -1,39 +1,70 @@
 {
   "name": "MiamiOH-shibboleth_idp",
-  "version": "1.3.0",
+  "version": "1.4.0",
   "author": "Dirk Tepe",
   "summary": "Manage Shibboleth IdP",
-  "license": "GPL-3.0+",
```
Comment (edestecd) [inline metadata.json]: DId you or pdk remove this +

Comment (edestecd) [reply metadata.json]: seems like GPL-3.0+ is depreciated

https://spdx.org/licenses/

Comment (edestecd) [reply metadata.json]: This is the new equiv:
GPL-3.0-or-later

File: metadata.json [hunk 7]
```diff
@@ -1,39 +1,70 @@
...
-  ]
+  ],
+  "tags": [
+    "shibboleth",
+    "shib",
+    "idp"
+  ],
+  "pdk-version": "1.17.0",
+  "template-url": "https://github.com/puppetlabs/pdk-templates#1.9.0",
+  "template-ref": "tags/1.9.0-0-g7281db5"
```
Comment (edestecd) [inline metadata.json]: didn't we find out that 1.9.0 tested puppet 5 and 6 in one of travis or gitlab ci?

Did we find a new version that tested puppet 4 and 5 in both?

File: manifests/install.pp (hunk 2 above)
Comment (edestecd) [reply manifests/install.pp]: I menat the new way to validate. With the Type in front of the parameter. See the next coment

Comment (kobybr) [reply manifests/install.pp]: I need to ensure casclient_source isn't empty if include_cas is true.   Otherwise I don't care what the value is

Comment (edestecd) [reply manifests/install.pp]: OK I understand now. I'm not sure there is a way to do that with the type validation.
Probably assert_type in an if

File: manifests/init.pp (hunk 3 above)
Comment (edestecd) [reply manifests/init.pp]: Didn't understand that you needed this only if something else was set.

File: manifests/params.pp (hunk 5 above)
Comment (kobybr) [reply manifests/params.pp]: using checksum_url in archive in lieu of

File: metadata.json (hunk 6 above)
Comment (kobybr) [reply metadata.json]: I removed due to deprecation,  will update to suggested

File: metadata.json (hunk 7 above)
Comment (kobybr) [reply metadata.json]: 1.9.1 does 5/6, 1.9.0 is what seems to work

**Comment by edestecd** [comment]
@kobybr looks good to me. Could you add at least one entry to CHANGELOG about this new version? Other than that I can mash the merge

**Pull request merged**